ql/src/java/org/apache/hadoop/hive/ql/exec/repl/incremental/IncrementalLoadTasksBuilder.java

```diff
@@ -95,6 +97,21 @@ public IncrementalLoadTasksBuilder(String dbName, String tableName, String loadP
     this.log.debug("Iteration num " + numIteration);
     TaskTracker tracker = new TaskTracker(conf.getIntVar(HiveConf.ConfVars.REPL_APPROX_MAX_LOAD_TASKS));
 
+    // if no events are there to replay, then update the last repl id of the database/ table to last event id.
+    if (!iterator.hasNext()) {
```

Comment:
There is already a check for iterator.hasNext() after while loop. Shall add this handling if taskChainTail is null in the else flow.

Reply:
taskChainTail is not null its initialized to taskChainhead and we can judge if while loop is exited with some task or not by putting one more check ..so both ways its same

Reply:
We can check for taskChainTail == evTaskRoot in the else flow after while loop which cover this case.

Reply:
done